docs/content/tutorials/nginx-dynamic-module.md

```diff
@@ -0,0 +1,115 @@
+- Load the module using a `configmap` into NGINX Ingress controller
+
+In order to build a custom NGINX Ingress image with specific modules, the first step you need to take is to modify the `Dockerfile` located in the `./build` directory of the NGINXINC repo.
+
+Clone the NGINX Ingress controller repo:
+
+```
+git clone git@github.com:nginxinc/kubernetes-ingress.git
+```
+
+Once you have the image cloned, edit the `Dockerfile` located in the `build` directory of the cloned repo.
+
+
+In our example, we are going to add the `Headers-more` dynamic module to our NGINX Ingress controller image.
+
+Locate your preferred OS version that you want to modify in the DOckerfile (debian, alpine etc.).
+Here is a snippet of the `Dockerfile` for `debian-plus`
+
+```
+FROM debian:${DEBIAN_VERSION} AS debian-plus
+ARG IC_VERSION
+ARG NGINX_PLUS_VERSION
+ARG BUILD_OS
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl apt-transport-https libcap2-bin \
+	&& curl -fsSL https://cs.nginx.com/static/keys/nginx_signing.key | gpg --dearmor > /etc/apt/trusted.gpg.d/nginx_signing.gpg \
+	&& curl -fsSL -o /etc/apt/apt.conf.d/90pkgs-nginx https://cs.nginx.com/static/files/90pkgs-nginx \
+	&& DEBIAN_VERSION=$(awk -F '=' '/^VERSION_CODENAME=/ {print $2}' /etc/os-release) \
+	&& printf "%s\n" "Acquire::https::pkgs.nginx.com::User-Agent \"k8s-ic-$IC_VERSION${BUILD_OS##debian-plus}-apt\";" >> /etc/apt/apt.conf.d/90pkgs-nginx \
+	&& printf "%s\n" "deb https://pkgs.nginx.com/plus/${NGINX_PLUS_VERSION^^}/debian ${DEBIAN_VERSION} nginx-plus" > /etc/apt/sources.list.d/nginx-plus.list \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y nginx-plus nginx-plus-module-njs \
+	&& apt-get purge --auto-remove -y apt-transport-https gnupg curl \
+	&& rm -rf /var/lib/apt/lists/*
+```
+
+Look for a line similar to the following line in the `Dockerfile`:
+
+```
+apt-get install --no-install-recommends --no-install-suggests -y nginx-plus nginx-plus-module-njs
+```
+
+This is the line you will want to modify/add the module you want to have loaded into NGINX Ingress controller.
+We are going to add the `headers-more` module. The updated line would look like this:
+
+```
+apt-get install --no-install-recommends --no-install-suggests -y nginx-plus nginx-plus-module-njs nginx-plus-module-headers-more
+```
+
+In the above example, I added a single module line:
+```
+nginx-plus-module-headers-more
+```
+
+After the new NGINX Ingress module image has been built successfully, the next step is to load the module into your NGINX Ingress controller when it will be deployed into your Kubernetes cluster.
```

Comment:
```suggestion
After the new NGINX Ingress module image has been built successfully, the next step is to load the module into your NGINX Ingress Controller when it will be deployed into your Kubernetes cluster.
```